### PR TITLE
Create NFTs components

### LINKED ISSUE #2634: Use new NFTs collection components on Wallet > NFTs tab
NFTs tab on Wallet page should stay the same as it is right now but it should use new NFTs Collection components.
This list should show NFTs for selected account on selected network - it's the same list as on the NFTs page but with predefined account and network filter.

### PR DESCRIPTION
**Pull request by jagodarybacka**
Resolves #2634

### What
This PR is all about the new NFTs list.

- added a bunch of useful selectors for the NFT redux slice
- updated types and what is being fetched from providers
- added a basic logic of fetching NFTs from collections on demand - to be improved in the future
- created all components that are building blocks for NFT list - expandable collection, single NFT item, preview

A lot of improvements, implementing missing pieces and polishing are still needed but this will unblock other developers using redux slice.

### Testing

- [ ] make sure all is working as before with `SUPPORT_NFT_TAB=false`
- [ ] checkout new NFT list in Wallet > NFTs - it won't work perfectly but expected is: displaying some collections owned by current account on the current network, fetching NFTs from collections on demand, displaying NFT preview

image
image

Latest build: [extension-builds-2667](https://github.com/tallycash/extension/suites/9564964787/artifacts/455308038) (as of Tue, 29 Nov 2022 14:57:31 GMT).

### CODE REVIEW

**Comment by hyphenized** [comment]
Nice work, it's looking good! Some initial comments: 
- Some of the NFT previews are very heavy (30Mb~), we should use simple hash sized previews when available to limit the amount of bandwidth we're using to load thumbnails/media and the resources used by the browser to render them.
- The [API response](https://api.simplehash.com/api/v0/nfts/owners?chains=ethereum&wallet_addresses=0xd8da6bf26964af9d7eed9e03e53415d37aa96045&collection_id=003b5a188936918e8015ee07068738c2) for some NFTs does not contain name/description/media properties. I think this might be the case for a lot of spam tokens so we should probably add a fallback for these cases or filter them out altogether
- I'm seeing some performance issues when scrolling/expanding collections that we might be able to solve with virtual lists, since, they are most likely coming from rendering a large number of items.

**Comment by jagodarybacka** [comment]
@hyphenized 

> NFT previews are very heavy
> performance issues when scrolling/expanding collections

Performance issues are waiting to be done later this sprint, let's not block this PR. I will probably need your help with it 😊 

> this might be the case for a lot of spam tokens

Lack of name or any other props is not the signal that this NFT is a spam, SH is indexing NFTs with a delay (after they are minted) so for example a lot of Galxe NFTs don't have image or title right after the user mints them. But they are visible in the wallet immediately and user can open them on Galxe. 
Follow up PR will fallback to "no title" and correct "no preview" img.

**Review by hyphenized — CHANGES_REQUESTED** [review]
Looks amazing 💯! Can't wait to ship this to our users! 

>Lack of name or any other props is not the signal that this NFT is a spam, SH is indexing NFTs with a delay (after they are minted) so for example a lot of Galxe NFTs don't have image or title right after the user mints them. But they are visible in the wallet immediately and user can open them on Galxe.

I see, appreciate the explanation. Left some comments and questions

**Review by kkosiorowska — COMMENTED** [review]
It's looking amazing! :fire: Just some non-blocking comments.

File: ui/_locales/en/messages.json [hunk 1]
```diff
@@ -230,6 +230,21 @@
     "empty": "Looks like you don't have any NFTs yet? Get some and view them here!",
     "noTitle": "No title",
     "NFTPricingComingSoon": "Coming soon: NFT price + sending",
+    "collectionHover": {
+      "close": "close",
+      "expand": "expand collection",
+      "view": "view details"
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/_locales/en/messages.json:236]: yup, that's the design in Figma
image

File: ui/components/Shared/SharedNetworkIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}): ReactElement {
+  const { network, size, hasBackground = false, backgroundOpacity = 1 } = props
+  return (
+    <>
+      <div className="icon_network_wrapper">
+        {hasBackground && <div className="icon_network_background" />}
+        <div className="icon_network" />
+      </div>
+      <style jsx>{`
+        .icon_network_wrapper {
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/Shared/SharedNetworkIcon.tsx:18]: yeah as we have a version with and without a transparent background

File: background/redux-slices/nfts_update.ts [hunk 3]
```diff
@@ -17,38 +21,67 @@ export type NFTsState = {
...
+    thumbnail,
+  } = collection
+  const { chainID } = network
+  acc.nfts[chainID] ??= {}
+  acc.nfts[chainID][owner] ??= {}
+  acc.nfts[chainID][owner][collection.id] = {
+    id,
+    name,
+    nftCount,
+    nfts: [],
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply background/redux-slices/nfts_update.ts:55]: from redux point of view - NFTs should be added to collection and removed as needed - basically I want to treat redux as a cache. Removing NFTs no longer owned is not yet implemented. To be improved in this sprint

File: ui/components/Shared/SharedSlideUpMenu.tsx [hunk 4]
```diff
@@ -130,6 +136,21 @@ export default function SharedSlideUpMenu(props: Props): ReactElement {
             opacity: 0;
             pointer-events: none;
           }
+          .slide_up_close {
+            position: sticky;
+            z-index: 2;
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/Shared/SharedSlideUpMenu.tsx:141]: we are not using them often but I agree :)

File: ui/components/NFTS_update/NFTItem.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          />
+          {!!floorPrice && (
+            <div className="nft_item_price">
+              ~{`${floorPrice.value} ${floorPrice.tokenSymbol}`}
+            </div>
+          )}
+        </div>
+        <NFTHover
+          onClick={() => onClick(item)}
+          isCollection={isCollection}
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/NFTS_update/NFTItem.tsx:38]: but then we would need whole icons object from NFTHover in the component above (NFTItem) as this is both icon but also label, background, size etc

File: ui/components/NFTS_update/NFTItem.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+export default function NFTsItem<T extends NFT | NFTCollectionCached>(props: {
+  item: T
+  isCollection?: boolean
+  isExpanded?: boolean
+  onClick: (value: T) => void
+}): ReactElement {
+  const { onClick, isCollection = false, isExpanded = false, item } = props
+  const { name = "No title", network, thumbnail } = item
+  const floorPrice =
+    "floorPrice" in item && item.floorPrice?.value && item.floorPrice
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/NFTS_update/NFTItem.tsx:17]: will add a function when I will write this 3th time :P

File: ui/components/NFTS_update/NFTList.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import NFTPreview from "./NFTPreview"
+
+export default function NFTList(props: {
+  collections: NFTCollectionCached[]
+}): ReactElement {
+  const { collections } = props
+  const [isPreviewOpen, setIsPreviewOpen] = useState(false)
+  const [currentNFTPreview, setCurrentNFTPreview] = useState<{
+    nft: NFT
+    collection: NFTCollectionCached
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply ui/components/NFTS_update/NFTList.tsx:15]: good catch!

**Review by hyphenized — APPROVED** [review]
Looks good!

**Pull request merged**